krm-functions/lib/fn-runtime/upstream_inventory_test.go

```diff
@@ -0,0 +1,539 @@
+	for _, tt := range test1 {
+		currentGVKN := &corev1.ObjectReference{
+			APIVersion:      tt.input.apiVersion,
+			Kind:            tt.input.kind,
+			Name:            tt.input.name,
+			Namespace:       tt.input.dummy,
+			FieldPath:       tt.input.dummy,
+			ResourceVersion: tt.input.apiVersion,
+		}
+		ipa := &nadv1.NetworkAttachmentDefinition{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: tt.input.apiVersion,
+				Kind:       tt.input.kind,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: tt.input.name,
+			},
+			Spec: nadv1.NetworkAttachmentDefinitionSpec{
+				Config: tt.input.spec,
+			},
+		}
+		byteStream, _ := yaml.Marshal(ipa)
+		kubeObjectMade, _ := fn.ParseKubeObject(byteStream)
+
+		ipa = &nadv1.NetworkAttachmentDefinition{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: tt.input.apiVersion,
+				Kind:       tt.input.kind,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: tt.input.name,
+			},
+			Spec: nadv1.NetworkAttachmentDefinitionSpec{
+				Config: "type2",
+			},
+		}
+		byteStream, _ = yaml.Marshal(ipa)
+		kubeObjectMade, _ = fn.ParseKubeObject(byteStream)
+		inventory.AddNewResource(currentGVKN, kubeObjectMade)
+
+	}
+	diffList, _ := inventory.Diff()
+	if len(diffList.CreateConditions) != 2 {
+		t.Errorf("TestDiffWithSpecToAddCondition: -want 2, +got:\n%v", len(diffList.CreateConditions))
+	}
+}
+
+func TestDiffWithSpecToDeleteCondition(t *testing.T) {
+	inventory := NewUpstreamInventory()
+	type object struct {
+		apiVersion string
+		kind       string
+		name       string
+		dummy      string
+		spec       string
+	}
+
+	test1 := []struct {
+		input object
+	}{
+		{
+			input: object{
+				apiVersion: "a",
+				kind:       "b",
+				name:       "c",
+				dummy:      "sample",
+				spec:       "test1",
+			},
+		},
+		{
+			input: object{
+				apiVersion: "d",
+				kind:       "e",
+				name:       "f",
+				spec:       "test2",
+				dummy:      "sample",
+			},
+		},
+	}
+	for _, tt := range test1 {
+		currentGVKN := &corev1.ObjectReference{
+			APIVersion:      tt.input.apiVersion,
+			Kind:            tt.input.kind,
+			Name:            tt.input.name,
+			Namespace:       tt.input.dummy,
+			FieldPath:       tt.input.dummy,
+			ResourceVersion: tt.input.apiVersion,
+		}
+		ipa := &nadv1.NetworkAttachmentDefinition{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: tt.input.apiVersion,
+				Kind:       tt.input.kind,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: tt.input.name,
+			},
+			Spec: nadv1.NetworkAttachmentDefinitionSpec{
+				Config: tt.input.spec,
+			},
+		}
+		byteStream, _ := yaml.Marshal(ipa)
+		kubeObjectMade, _ := fn.ParseKubeObject(byteStream)
+		inventory.AddExistingCondition(currentGVKN, &kptv1.Condition{
+			Type:    currentGVKN.APIVersion + currentGVKN.Kind + currentGVKN.Name,
+			Status:  kptv1.ConditionFalse,
+			Reason:  tt.input.dummy,
+			Message: tt.input.dummy,
+		})
+		inventory.AddExistingResource(currentGVKN, kubeObjectMade)
+
+		ipa = &nadv1.NetworkAttachmentDefinition{
```

Comment:
This block not needed as we are not adding new resource

Reply:
@ganchandrasekaran can you look at this?

Reply:
Fixed to add it using only single block